authorizers/iconik/authorizer.ts

```diff
@@ -0,0 +1,88 @@
+import { IconikService } from '@workflowwin/iconik-api';
+import { CustomActionPayload } from '@workflowwin/iconik-api/dist/src/assets/assets-methods';
+import { AppError, CommonErrors } from '@helper/app-error';
+import { getEnv } from '@helper/environment';
+import { log } from '@helper/logger';
+import { IconikContext } from './context';
+
+type Payload = CustomActionPayload & { auth_token?: string };
+
+/**
+ * Iconik provides the auth token in request body, but custom authorizers don`t support request body.
+ * So we need to check it in the Lambda function instead using API Gateway authorizers
+ * How to use this authorizer:
+ * const authContext = await authorizeWithIconikCustomAction(event.body);
+ *
+ * Then you can create IconikService with User's Auth Context or App Credentials stored in env
+ * const iconik = createIconikClient(authContext);
+ * or
+ * const iconikClient = createIconikClient();
+ *
+ * @param {Payload} customActionBody
+ * @returns {Promise<IconikContext>}
+ */
+
+export async function authorizeWithIconikCustomAction(customActionBody: Payload): Promise<IconikContext> {
+  const { user_id: callerId, system_domain_id: systemDomainId, auth_token: authToken } = customActionBody;
+
+  const iconikUrl = getEnv('ICONIK_URL');
+  const appId = getEnv('ICONIK_APP_ID');
+  const requiredIconikDomainId = getEnv('ICONIK_DOMAIN_ID');
```

Comment:
I have faced trouble with getting the variable's value when I tried to use destructuring on process.env(as I remember it worked before). So, I checked getEnv function in deployed lambda function:
- helper/environment.ts was copied and I added APP_ID to envs object,
- used same versions envs libraries and serverless-env-generator.
I got the following results:
Lambda code:
```
export const testEnv: Handler<void> = async (event) => {
  try {
    const envs = ['APP_ID', 'TOKEN_ADMIN', 'ICONIK_URL'];
    const res = {};
    console.log('getStage', getStage());
    console.log('isStage dev', isStage('dev'));
    for (const env of envs) {
      const qe = getEnv(env);
      console.log(env, qe);
    }
    const getAppId = getEnv('APP_ID');
    console.log('APP IDs', getAppId, process.env.APP_ID);
    return res;
  } catch (error) {
    console.error(error);
  }
```
Logic for STAGE and APP_ID works correctly, but when I tried to extract other variables I got undefined. 
![image](https://user-images.githubusercontent.com/45095318/113923137-d9fb4380-97f0-11eb-80b8-096c12cefbcc.png)
I think we cannot get value from process.env by this construction:
```
 process.env[name];
```

Reply:
It doesn\`t work because you don\`t extract env variables into the `envs` object here
![image](https://user-images.githubusercontent.com/69796613/113958172-4a25bb80-9829-11eb-969d-9508d0fa3970.png)
We use `webpack-dotenv` and we need explicitly extract all env variables defined in `env.yml`.

Reply:
I will think about how to do this automatically(may be another webpack plugin=))

Reply:
I updated the message to include the limitations of dotenv-webpack.

Reply:
I think we can enhance it later. Or you can continue to use the old way like `process.env.VAR!`.

Reply:
I am about that getEnv contains ```process.env[name]``` construction and it can be confusing. If someone will try to use this function to get variables from process.env and they will get undefined.

Reply:
`process.env[name]` used to get variables that was defined in `serverless.yml` and available in runtime.